src/devices/Mhz19b/Mhz19b.cs

```diff
@@ -0,0 +1,256 @@
+        /// </summary>
+        /// <param name="stream">Existing stream</param>
+        /// <param name="shouldDispose">If true, the stream gets disposed when disposing the binding</param>
+        public Mhz19b(Stream stream, bool shouldDispose)
+        {
+            _serialPortStream = stream ?? throw new ArgumentNullException(nameof(stream));
+            _shouldDispose = shouldDispose;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Mhz19b"/> class and creates a new serial port stream.
+        /// </summary>
+        /// <param name="uartDevice">Path to the UART device / serial port, e.g. /dev/serial0</param>
+        /// <exception cref="System.ArgumentException">uartDevice is null or empty</exception>
+        public Mhz19b(string uartDevice)
+        {
+            if (string.IsNullOrEmpty(uartDevice))
+            {
+                throw new ArgumentException(nameof(uartDevice));
+            }
+
+            // create serial port using the setting acc. to datasheet, pg. 7, sec. general settings
+            _serialPort = new SerialPort(uartDevice, 9600, Parity.None, 8, StopBits.One)
+            {
+                Encoding = Encoding.ASCII,
+                ReadTimeout = 1000,
+                WriteTimeout = 1000
+            };
+            _serialPort.Open();
+            _serialPortStream = _serialPort.BaseStream;
+            _shouldDispose = true;
+        }
+
+        /// <summary>
+        /// Gets the current CO2 concentration from the sensor.
+        /// The validity is true if the current concentration was successfully read.
+        /// If the serial communication timed out or the checksum was invalid the validity is false.
+        /// If the validity is false the ratio is set to 0.
+        /// </summary>
+        /// <returns>CO2 concentration in ppm and validity</returns>
+        /// <exception cref="System.IO.IOException">Communication with sensor failed</exception>
+        public VolumeConcentration GetCo2Reading()
+        {
+            try
+            {
+                // send read command request
+                var request = CreateRequest(Command.ReadCo2Concentration);
+                request[(int)MessageFormat.Checksum] = Checksum(request);
+                _serialPortStream.Write(request, 0, request.Length);
+
+                // read complete response (9 bytes expected)
+                byte[] response = new byte[MessageSize];
+                int timeout = 100;
+                int bytesRead = 0;
+                while (timeout > 0 && bytesRead < MessageSize)
+                {
+                    bytesRead += _serialPortStream.Read(response, bytesRead, response.Length - bytesRead);
+                    Thread.Sleep(1);
+                    timeout--;
+                }
+
+                if (timeout == 0)
+                {
+                    throw new IOException("Timeout");
```

Comment:
TimeoutException?

Reply:
if this is relatively probable event then you should change method to `bool TryGetCo2Reading(out VolumeConcentration volumeConcentration)`

Reply:
No, this doesn't happen relatively probable. Actually I never saw this except when my test system was some how heavily loaded or the hardware was somehow disturbed. I don't see the need for a Try...-method here.

I'll change the type of the exception. I had a broader sematic in mind when I chose IOException, but eventually it is a timeout and there is no further root cause available and guessing doesn't help.

Reply:
Done